File: structural/bridge/python/concrete_implementator.py
from implementator import Device


class TV(Device):
    def __init__(self):
        self._power_on = False

    def on(self):
        self._power_on = True
        print("TV is now ON")

    def off(self):
        self._power_on = False
        print("TV is now OFF")

    def is_on(self) -> bool:
        return self._power_on


class Radio(Device):
    def __init__(self):
        self._power_on = False

    def on(self):
        self._power_on = True
        print("Radio is now ON")

    def off(self):
        self._power_on = False
        print("Radio is now OFF")

    def is_on(self) -> bool:
        return self._power_on

File: structural/bridge/typescript/concrete-implementator.ts
import { Device } from "./implementor";

export class TV implements Device {
	private powerOn: boolean = false;

	on(): void {
		this.powerOn = true;
		console.log('TV is now ON');
	}

	off(): void {
		this.powerOn = false;
		console.log('TV is now OFF');
	}

	isOn(): boolean {
		return this.powerOn;
	}
}

export class Radio implements Device {
	private powerOn: boolean = false;

	on(): void {
		this.powerOn = true;
		console.log('Radio is now ON');
	}

	off(): void {
		this.powerOn = false;
		console.log('Radio is now OFF');
	}

	isOn(): boolean {
		return this.powerOn;
	}
}
